railties/lib/rails/generators/rails/plugin/templates/bin/rails.tt

```diff
@@ -3,6 +3,7 @@
 
 ENGINE_ROOT = File.expand_path('../..', __FILE__)
 ENGINE_PATH = File.expand_path('../../lib/<%= namespaced_name -%>/engine', __FILE__)
+APP_PATH = File.expand_path('../../<%= dummy_path -%>/config/application', __FILE__)
```

Comment:
Is the `APP_PATH` trickery in the `rails/engine/commands` file to provide backwards compat for this line?

Won't the `rails app:update` command add this line for users?